packages/intrepid/matlab/intrelab/CMakeLists.txt

```diff
@@ -39,159 +55,20 @@ endif()
 
 function(make_mex fname)
-
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/objs/${fname}.o
-      COMMAND "${MEX}" "LDFLAGS=-shared" "CXX=${CMAKE_CXX_COMPILER}" "CXXFLAGS=${MEX_FLAGS}" "-largeArrayDims" -c ${Trilinos_INCLUDE_DIRS} ${Trilinos_TPL_INCLUDE_DIRS} "${CMAKE_CURRENT_SOURCE_DIR}/src/${fname}.cpp"
-      BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/objs/${fname}.o
-      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/src/${fname}.cpp
-      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/objs
-      )
-  
-  #add_custom_command(OUTPUT ${fname}.${MEX_EXTENSION}
-  #    COMMAND "${MEX}" "LDFLAGS=-shared" "CXX=${CMAKE_CXX_COMPILER}" "CXXFLAGS=${MEX_FLAGS}" "-largeArrayDims" ${Trilinos_INCLUDE_DIRS} ${Trilinos_TPL_INCLUDE_DIRS} ${Trilinos_LIBRARY_DIRS} ${Trilinos_TPL_LIBRARY_DIRS} ${l_Trilinos} ${l_Trilinos_TPL} "${CMAKE_CURRENT_BINARY_DIR}/objs/${fname}.o" "${CMAKE_CURRENT_BINARY_DIR}/objs/m2i_helpers.o"
-  #    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/objs/${fname}.o ${CMAKE_CURRENT_BINARY_DIR}/objs/m2i_helpers.o
-  #    BYPRODUCTS ${fname}.${MEX_EXTENSION}
-  #    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  #    )
-
-  add_custom_command(OUTPUT ${fname}.${MEX_EXTENSION}
-      COMMAND "${MEX}" "${MEX_LD_FLAGS}" "CXX=${CMAKE_CXX_COMPILER}" "CXXFLAGS=${MEX_FLAGS}" "-largeArrayDims" ${Trilinos_INCLUDE_DIRS} ${Trilinos_TPL_INCLUDE_DIRS} ${Trilinos_LIBRARY_DIRS} ${Trilinos_TPL_LIBRARY_DIRS} ${l_Trilinos} ${l_Trilinos_TPL} "${CMAKE_CURRENT_SOURCE_DIR}/src/${fname}.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/src/m2i_helpers.cpp"
-      #DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/objs/${fname}.o ${CMAKE_CURRENT_BINARY_DIR}/objs/m2i_helpers.o
-      BYPRODUCTS ${fname}.${MEX_EXTENSION}
-      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-      )
-  
-  install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/${fname}.${MEX_EXTENSION} DESTINATION ${CMAKE_INSTALL_PREFIX})
-
+  matlab_add_mex(NAME ${fname}
+      SHARED
+      SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/m2i_helpers.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/${fname}.cpp
+  )
+  target_link_libraries(${fname} Trilinos::all_libs)
 endfunction(make_mex)
```

Comment:
It is actually considered bad CMake form to repeat the function name in `endfunction(<funcname>)` and is only supported for backward compatibility.  Just write `endfunction()`.  See:

* https://cmake.org/cmake/help/v3.17/command/endfunction.html